File: microkernel/api/src/main/java/ink/reactor/kernel/logger/LoggerBuilder.java
package ink.reactor.kernel.logger;

import ink.reactor.kernel.Reactor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public final class LoggerBuilder {
    private final LoggerFactory factory;

    public LoggerBuilder() {
        this.factory = Reactor.get().loggerFactory();
    }

    private String prefix, suffix;
    private LoggerFormatter formatter;

    public LoggerBuilder withPrefix(final String prefix) {
        this.prefix = prefix;
        return this;
    }

    public LoggerBuilder withSuffix(final String suffix) {
        this.suffix = suffix;
        return this;
    }

    public LoggerBuilder withFormatter(final LoggerFormatter formatter) {
        this.formatter = formatter;
        return this;
    }

    public Logger build() {
        return factory.createLogger(this);
    }
}

File: kernel/api/src/main/kotlin/ink/reactor/kernel/logger/LoggerBuilder.kt
package ink.reactor.kernel.logger

import ink.reactor.kernel.Reactor

class LoggerBuilder(
    private val factory: LoggerFactory = Reactor.loggerFactory
) {

    var prefix: String = ""
    var suffix: String = ""
    var formatter: LoggerFormatter? = null

    fun withPrefix(prefix: String): LoggerBuilder {
        this.prefix = prefix
        return this
    }

    fun withSuffix(suffix: String): LoggerBuilder {
        this.suffix = suffix
        return this
    }

    fun withFormatter(formatter: LoggerFormatter): LoggerBuilder {
        this.formatter = formatter
        return this
    }

    fun build(): Logger {
        return factory.createLogger(this)
    }
}
